### ISSUE TITLE
Exploit crash WS server after sending empty message from client

### ISSUE THREAD

**Issue by anboo** [GeniusesOfSymfony/WebSocketBundle#227]
https://gist.github.com/anonymous/00d5f0f7b2b87581f53179c09924d264 - my topic, security.yml, config.yml, pubsub/routing.yml

Usually, everything is fine, but sometimes throws an exception

> [2017-03-15 21:32:21] websocket.DEBUG: GET CLIENT 1177  
> [2017-03-15 21:34:53] websocket.DEBUG: GET CLIENT 1172  
> [2017-03-15 21:34:53] websocket.INFO: admin publish to chat/1  
> [2017-03-15 21:34:53] websocket.DEBUG: Matched route "chat.chat_topic" {"room_id":"1"} 
> 
>                                                                                
>   [Symfony\Component\Debug\Exception\FatalThrowableError]                      
>   Type error: Too few arguments to function ChatBundle\Topic\ChatTopic::onPub  
>   lish(), 3 passed in /home/anboo/www/kwork/vendor/gos/web-socket-bundle/Serv  
>   er/App/Dispatcher/TopicDispatcher.php on line 162 and exactly 6 expected

**Comment by anboo** [comment]
I understood the problem. This happens when I send empty string message: `session.publish('chat/topic1', '')` and further this code ceases to define the request as:

```
if ($payload) { //its a publish call.
    $appTopic->{$calledMethod}($conn, $topic, $request, $payload, $exclude, $eligible);
} else {
    $appTopic->{$calledMethod}($conn, $topic, $request);
}
```

But can not exclude the hit of an empty string on the client, because This check can be removed by a hackers.

**Comment by anboo** [comment]
$playload can take the value "" (empty string) or NULL (not publish method, not found playload).
This requires strict type checking, please, fix it as quickly as possible. This is a very serious bug. 
And so your bundle is a terrific thing!

**Comment by ProPheT777** [comment]
The fix has been released in v1.8.7, thank you @anboo
